### PR TITLE
Fix `Vasprun.get_potcars` search method; tweak fake POTCARs

### PR DESCRIPTION
**Pull request by esoteric-ephemera**
The main change here is to correct the `update_potcar_spec` method of `Vasprun` found by @Andrew-S-Rosen in issue #3586. Specifying `Vasprun("vasprun.xml")` would not find POTCARs, but specifying the leading `Vasprun("./vasprun.xml")` would. That has been corrected.

Other fixes:
- Add test for this behavior, `io/test_outputs.TestVasprun.test_parse_potcar_cwd_relative`
- Update potcar scrambling to ensure that scrambled POTCARs have same TITELs as originals. A "FAKE" comment still exists in the scrambled POTCARs to show that they're gobbled, but this is no longer parsed into `PotcarSingle.TITEL`

### CODE REVIEW

File: pymatgen/io/vasp/inputs.py [hunk 1]
```diff
@@ -272,6 +272,7 @@ def from_file(cls, filename, check_for_potcar=True, read_velocities=True, **kwar
                     [get_el_sp(n) for n in names]  # ensure valid names
                 except Exception:
                     names = None
+        print("shitfucker", names, dirname, filename)
```
Comment (janosh) [inline pymatgen/io/vasp/inputs.py:275]: my favorite line in this PR! 😄

**Review by janosh — APPROVED** [review]
thanks @esoteric-ephemera for the quick fix!

File: pymatgen/io/vasp/inputs.py (hunk 1 above)
Comment (Andrew-S-Rosen) [reply pymatgen/io/vasp/inputs.py]: LOL I am losing it over here.

Comment (esoteric-ephemera) [reply pymatgen/io/vasp/inputs.py]: Ahhhhhh I'm so sorry! I was maybe working a bit too fast lol

**Pull request merged**